### PR TITLE
refactor(optimizer): change "filter_bias_and_bn" to "weight_decay_filter"

### PR DESCRIPTION
**Pull request by sageyou**
Thank you for your contribution to the MindCV repo.
Before submitting this PR, please make sure:

- [x] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md)
- [x] Your code builds clean without any errors or warnings
- [x] You are using approved terminology
- [x] You have added unit tests

## Motivation

Fix the bug with weight decay when set `filter_bias_and_bn`:

1. When `filter_bias_and_bn` is `True`: function `init_group_params` does not set value of weight decay for `no_decay_params`, in this case, mindspore will use `weight_decay` of optimizer (usually not 0.0).
2. When `filter_bias_and_bn` is `False`: mindspore will automatically filter BatchNorm params from weight_decay.

So the name of the argument is not the same as what it actually does. And we can never filter out the param of bias and norm layer from doing weight decay, as the name `filter_bias_and_bn`.

Due to this, we refactor it to `weight_decay_filter`:
- `"disable"`: No parameters to filter.
- `"auto"`: We do not apply weight decay filtering to any parameters. However, MindSpore currently automatically filters the parameters of Norm layer from weight decay.
- `"norm_and_bias"`: Filter the paramters of Norm layer and Bias from weight decay.

How do I migrate from an old configuration?
| filter_bias_and_bn | weight_decay_filter |
|  ----  | ----  |
| `True` | `"disable"` |
| `False` | `"auto"` |

> Keep in mind that filter_bias_and_bn doesn't do what its name suggests.

BTW, we also support get no_weight_decay list from model and layer_decay.

## Test Plan

(How should this PR be tested? Do you require special setup to run the test or repro the fixed bug?)

## Related Issues and PRs

(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

### CODE REVIEW

**Review by geniuspatrick — APPROVED** [review]

**Review by SamitHuang — APPROVED** [review]

File: mindcv/optim/optim_factory.py [hunk 1]
```diff
@@ -59,19 +219,52 @@ def create_optimizer(
             of current step. Default: 0.
         momentum: momentum if the optimizer supports. Default: 0.9.
         nesterov: Whether to use Nesterov Accelerated Gradient (NAG) algorithm to update the gradients. Default: False.
-        filter_bias_and_bn: whether to filter batch norm parameters and bias from weight decay.
-            If True, weight decay will not apply on BN parameters and bias in Conv or Dense layers. Default: True.
+        weight_decay_filter: filters to filter parameters from weight_decay.
+            - "disable": No parameters to filter.
+            - "auto": We do not apply weight decay filtering to any parameters. However, MindSpore currently
+                    automatically filters the parameters of Norm layer from weight decay.
```
Comment (SamitHuang) [inline mindcv/optim/optim_factory.py:225]: Be specific. For 'auto' weight decay, `Norm layer` , which kind of norm layers will be applied with weight decay? BatchNorm only or including LayerNorm

Comment (sageyou) [reply mindcv/optim/optim_factory.py:225]: All the norm layers BatchNorm and LayerNorm.
